src/main/java/com/devonfw/ide/sonarqube/common/impl/check/thirdparty/DevonArchitecture3rdPartyImmutableCheck.java

```diff
@@ -0,0 +1,31 @@
+package com.devonfw.ide.sonarqube.common.impl.check.thirdparty;
+
+import com.devonfw.ide.sonarqube.common.api.JavaType;
+import com.devonfw.ide.sonarqube.common.impl.check.DevonArchitecture3rdPartyCheck;
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * {@link DevonArchitecture3rdPartyCheck} verifying that Immutable is used from the proper (org.hibernate.annotations) package.
+ */
+@Rule(key = "E8", name = "devonfw 3rd Party Immutable Check", //
+    priority = Priority.CRITICAL, tags = { "architecture-violation", "devonfw", "thirdparty" })
+public class DevonArchitecture3rdPartyImmutableCheck extends DevonArchitecture3rdPartyCheck {
+
+    private Set<String> allImports = new HashSet<>();
+
+    @Override protected String checkDependency(JavaType source, JavaType target) {
+
+        allImports.add(target.toString());
+        String sourceSimpleName = source.getSimpleName();
+        if (sourceSimpleName.contains("Entity") && allImports.contains("javax.annotation.concurrent.Immutable")
```

Comment:
I think the third condition checking if 'javax.persistence.Entity' is imported could be removed. Due to this condition, abstract entity classes, which do not have this import, will not be recognized.
'sourceSimpleName.contains("Entity")' is enough to detect Entity classes, especially when we consider that our plugin is only relevant for devon4j projects, where there are naming conventions.

Additionally I would get rid of the 'allImports' list and just check the import with 'target.toString().equals("javax.annotation.concurrent.Immutable")'. Your solution saves all the imports that were found during the analysis, which includes imports from classes that have already been analyzed. This will raise a lot of false positives, as one class containing the forbidden import will make this issue get raised on all following classes.